src/DetailsView/components/report-export-component.tsx

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { escape } from 'lodash';
+import { ActionButton } from 'office-ui-fabric-react/lib/Button';
+import * as React from 'react';
+
+import { NamedSFC } from '../../common/react/named-sfc';
+import { ExportResultType } from '../../common/telemetry-events';
+import { ReportGenerator } from '../reports/report-generator';
+import { ExportDialog, ExportDialogDeps } from './export-dialog';
+
+export interface ReportExportComponentProps {
+    deps: ExportDialogDeps;
+    exportResultsType: ExportResultType;
+    reportGenerator: ReportGenerator;
+    pageTitle: string;
+    scanDate: Date;
+    htmlGenerator: (descriptionPlaceholder: string) => string;
+}
+
+export const ReportExportComponent = NamedSFC<ReportExportComponentProps>('ReportExportComponent', props => {
+    const descriptionPlaceholder: string = 'd68d50a0-8249-464d-b2fd-709049c89ee4';
+    const [isOpen, setDialogVisibility] = React.useState(false);
```

Comment:
this is new to me: `React.useState` I'm guessing this are the new react hooks right?
I guess I see the point but it obscure the type of the state (there is literally to type for the state of this component).
I'm wondering if this deserve some engineering discussion before adopt the usage of hooks.
Thoughts?

Reply:
The type of the state could be more clear if I use 
`const [isOpen, setDialogVisibility] = React.useState<boolean>(false);`

Yeah, this is the first time I use hooks as well, I feel it is much more convenient. But yes, it is definitely worth a discussion. We can use this PR as a reference when we discuss it.

Reply:
I feel that Introducing React hooks here might be a little too much for this refactor; but thats just my opinion. I would be interested to know the viewpoint of the team on a spike or something for this.
Also, Just having hooks in one file gets us away from the consistency that we try to follow in every other part of the code. my 2 cents.

Reply:
Updated.